core/src/main/java/org/ehcache/spi/cache/Store.java

```diff
@@ -441,7 +441,7 @@
      * Informs this Provider, a Store it created is being disposed (i.e. closed)
      * @param resource
      */
-    void releaseStore(Store<?, ?> resource);
+    void releaseStore(Store<?, ?> resource) throws CacheAccessException;
```

Comment:
Right, and there is logic in those state transitions, so _I think_ in this case, the `.shutdown()` will continue. If anything though, I agree with @chrisdennis and we shouldn't make throwing `CacheAccessException` part of the contract, as it's an internal, implementation specific detail that the `Store` is even accessed further when `releaseStore` is invoked.